utils/field.h

```diff
@@ -9,6 +9,7 @@ struct field_data {
 	struct uftrace_task_reader *task;
 	struct fstack *fstack;
 	void *arg;
+	unsigned long addr;
```

Comment:
It's unnatural to add 'addr' field as it can be get from others.  Why do you make it special?  Also `unsigned long` is not sufficient since 32 bit system should run 64 bit data.